### PR TITLE
miru: darwin support, add maintainer

### PR DESCRIPTION
**Pull request by matteo-pacini**
## Description of changes

- Split the package into `linux` and `darwin`
- Kept `linux` as it is, and added relevant code for `darwin`
- Added myself as maintainer for Darwin

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mi/miru/package.nix [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
     license = licenses.gpl3Plus;
-    maintainers = [ maintainers.d4ilyrun ];
+    maintainers = with maintainers; [
+      d4ilyrun
+      matteopacini
+    ];
     mainProgram = "miru";
 
-    platforms = [ "x86_64-linux" ];
+    platforms = [ "x86_64-linux" ] ++ platforms.darwin;
```
Comment (d4ilyrun) [inline pkgs/by-name/mi/miru/package.nix:19]: Does it also work on `x86_64-darwin`?

**Review by d4ilyrun — COMMENTED** [review]
I cannot test it myself but I'll trust you and the CI 😄 
LGTM

File: pkgs/by-name/mi/miru/package.nix (hunk 1 above)
Comment (matteo-pacini) [reply pkgs/by-name/mi/miru/package.nix:19]: It does, it's an universal app bundle, i.e.:

image

Comment (d4ilyrun) [reply pkgs/by-name/mi/miru/package.nix:19]: Ok perfect then 😄

**Review by d4ilyrun — APPROVED** [review]

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 321539` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>miru</li>
  </ul>
</details>

**Review by 4evy — APPROVED** [review]

File: pkgs/by-name/mi/miru/darwin.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,Applications}
+    cp -r Miru.app $out/Applications/
+    ln -s $out/Applications/Miru.app/Contents/MacOS/Miru $out/bin/miru
+  '';
```
Comment (wegank) [inline pkgs/by-name/mi/miru/darwin.nix:26]:
```suggestion
  nativeBuildInputs = [
    unzip
    makeWrapper
  ];

  installPhase = ''
    runHook preInstall

    mkdir -p $out/{bin,Applications}
    cp -r Miru.app $out/Applications/
    makeWrapper $out/Applications/Miru.app/Contents/MacOS/Miru $out/bin/miru

    runHook postInstall
  '';
```

`$out/bin/miru` won't work as expected when created with `ln -s`.

**Comment by matteo-pacini** [comment]
@wegank addressed your comments, and tested out "$out/bin/miru", works perfectly, thanks for the suggestion.

Ready for re-review 🙏🏻

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 321539` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>miru</li>
  </ul>
</details>

**Review by 4evy — APPROVED** [review]

**Pull request merged**